Sources/ScipioKit/Resolver/ResolvedModels.swift

```diff
@@ -0,0 +1,233 @@
+import Foundation
+import PackageManifestKit
+
+typealias Pins = [PackageID: Pin.State]
+
+struct ModulesGraph {
+    var rootPackage: ResolvedPackage
+    var allPackages: [PackageID: ResolvedPackage]
+    var allModules: Set<ResolvedModule>
+
+    func package(for module: ResolvedModule) -> ResolvedPackage? {
+        allPackages[module.packageID]
+    }
+
+    func module(for name: String) -> ResolvedModule? {
+        allModules.first { $0.name == name }
+    }
+}
+
+struct PackageID: Hashable {
+    var description: String
+    var packageIdentity: String
+
+    init(
+        packageKind: PackageKind,
+        packageIdentity: String
+    ) {
+        self.packageIdentity = packageIdentity
+        switch packageKind {
+        case .root(let url), .fileSystem(let url), .localSourceControl(let url):
+            description = url.absoluteString
+        case .remoteSourceControl(let string):
+            description = string
+        case .registry(let string):
+            description = string
+        }
+    }
+}
```

Comment:
I think that the implementation will not have correct dependency mirroring support which should exist in the original SwiftPM implementation: https://github.com/swiftlang/swift-package-manager/blob/1cbb77aa98ab419c57d0d78287ac2a3a4202c1d7/Sources/PackageModel/IdentityResolver.swift#L25-L77

Removing dependency mirroring support will be a breaking change or may make the tool unable to use in some real-world use cases. Is it an intentional decision? Is it okay to remove that support?

@Ryu0118 @giginet

Reply:
It's better to note the concern for now. However, we needn't resolve this issue in this PR.

First, we should investigate whether `dump-package` contains mirroring information.